torch/csrc/autograd/init.cpp

```diff
@@ -52,10 +56,22 @@ PyObject* THPAutograd_initExtension(PyObject* _unused, PyObject *unused) {
   m.def("_enable_profiler", enableProfiler);
   m.def("_disable_profiler", disableProfiler);
   m.def("_profiler_enabled", profilerEnabled);
-  m.def("_run_before_callbacks", _runBeforeCallbacks);
 
-  py::class_<RecordFunction, std::shared_ptr<RecordFunction>>(m, "_RecordFunction")
-    .def(py::init<>());
+  // TODO: remove when jit future can hold PyObject (https://github.com/pytorch/pytorch/issues/34999)
+#ifdef USE_DISTRIBUTED
+  m.def(
+      "_call_end_callbacks_on_fut",
+      [](const at::Tensor& handle,
+         const std::shared_ptr<torch::distributed::rpc::FutureMessage>& fut) {
+        torch::autograd::profiler::_call_end_callbacks_on_fut(handle, fut);
+      });
+#else
+  m.def("_call_end_callbacks_on_fut", []() {
```

Comment:
It won't be possible to call this function without the distributed package since the python side would need a python instance of `torch::distributed::rpc::FutureMessage`, so I'm wondering if it's actually better to just leave this out entirely if `USE_DISTRIBUTED=0`.

Reply:
The issue with having the same arg type is that one of the args is `std::shared_ptr<torch::distributed::rpc::FutureMessage>`, and this arg type is only available if we are compiling with `USE_DISTRIBUTED`. So I'm not sure how to get the same arg type

Reply:
I think this might be better left out entirely, since the user should never have to call `rf._call_end_callbacks_on_fut` themselves on a future. This should be done transparently in the profiling logic we've added.

Reply:
I agree, let's leave this out.